### PR TITLE
ci: fix ci issues related to osd

### LINKED ISSUE #249: Fix intermittent CI issues
rbd command fails very frequently in CI  and restarting the CI fixes. There may be some corner cases but we need to fix those.

### PR DESCRIPTION
**Pull request by subhamkrai**
**Description of your changes:**
ci: fix ci issues related to osd

due to some flaky osd pod was not coming up
and ci tests were failing. Also, improving some
ci related code.

**Which issue is resolved by this Pull Request:**
Resolves #249 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] CI tests has been updated, if necessary.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.

### CODE REVIEW

File: tests/github-action-helper.sh [hunk 1]
```diff
@@ -74,13 +81,43 @@ deploy_with_custom_ns() {
...
+}
+
 wait_for_pod_to_be_ready_state_default() {
   timeout 200 bash <<-'EOF'
     until [ $(kubectl get pod -l app=rook-ceph-osd -n rook-ceph -o jsonpath='{.items[*].metadata.name}' -o custom-columns=READY:status.containerStatuses[*].ready | grep -c true) -eq 1 ]; do
       echo "waiting for the pods to be in ready state"
       sleep 1
     done
 EOF
+  if [ $? -eq 124 ]; then
```
Comment (travisn) [inline tests/github-action-helper.sh:117]: 124 is the error code from the timeout? maybe add a comment about that

File: tests/github-action-helper.sh [hunk 2]
```diff
@@ -6,11 +6,27 @@ set -xeEo pipefail
 # VARIABLES #
 #############
 : "${FUNCTION:=${1}}"
-: "${BLOCK:=$(sudo lsblk --paths | awk '/14G/ || /75G/ {print $1}' | head -1)}"
+
+function find_extra_block_dev() {
```
Comment (travisn) [inline tests/github-action-helper.sh:10]: This is the same function from the rook repo, right? Or any changes? How about adding a comment here for where the rook function is found in case future bug fixes are needed

File: tests/github-action-helper.sh [hunk 3]
```diff
@@ -74,13 +81,43 @@ deploy_with_custom_ns() {
...
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/filesystem-test.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/subvolumegroup.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/csi/rbd/storageclass-test.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/csi/rbd/pvc.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/csi/cephfs/pvc.yaml
+}
+
+deploy_csi_driver_custom_ns() {
+  curl https://raw.githubusercontent.com/rook/rook/master/deploy/examples/csi/rbd/storageclass-test.yaml -o storageclass-rbd-test.yaml
+  sed -i "s|provisioner: rook-ceph.rbd.csi.ceph.com |provisioner: test-operator.rbd.csi.ceph.com |g" storageclass-rbd-test.yaml
```
Comment (travisn) [inline tests/github-action-helper.sh:94]: The custom namespace is always `test-operator`? It might be nice to define that as a var somewhere (in a future PR), but maybe not worth the effort.

Comment (subhamkrai) [reply tests/github-action-helper.sh:94]: hmm will look at this in separate PR

File: .github/workflows/go-test.yaml [hunk 4]
```diff
@@ -68,11 +68,10 @@ jobs:
           kubectl -n rook-ceph wait deployment rook-ceph-mon-d --for condition=Available=True --timeout=90s
           kubectl -n rook-ceph wait deployment rook-ceph-mon-e --for condition=Available=True --timeout=90s
 
-      # TODO: To fix when running in CI
-      # - name: Rbd command
-      #   run: |
-      #     set -ex
-      #     kubectl rook-ceph rbd ls replicapool
+      - name: Rbd command
```
Comment (Madhu-1) [inline .github/workflows/go-test.yaml:71]: `Rbd` to `RBD` take care of this in other places also

File: tests/github-action-helper.sh [hunk 5]
```diff
@@ -74,13 +82,45 @@ deploy_with_custom_ns() {
   kubectl create -f "$3"
 }
 
+deploy_csi_driver_default_ns() {
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/filesystem-test.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/subvolumegroup.yaml
+  kubectl create -f https://raw.githubusercontent.com/rook/rook/master/deploy/examples/csi/rbd/storageclass-test.yaml
```
Comment (Madhu-1) [inline tests/github-action-helper.sh:88]: why we are not creating cephfs storageclass here?

File: tests/github-action-helper.sh [hunk 6]
```diff
@@ -99,6 +144,11 @@ wait_for_operator_pod_to_be_ready_state_default() {
       sleep 1
     done
 EOF
+  # timeout command return exit status 124 if command times out
+  if [ $? -eq 124 ]; then
```
Comment (Madhu-1) [inline tests/github-action-helper.sh:148]: define const for 124 and use it as its used in different places

File: tests/github-action-helper.sh [hunk 7]
```diff
@@ -143,6 +208,11 @@ wait_for_crd_to_be_ready_custom() {
       sleep 2
     done
 EOF
+  # timeout command return exit status 124 if command times out
+  if [ $? -eq 124 ]; then
```
Comment (Madhu-1) [inline tests/github-action-helper.sh:212]: if possible move this to helper function so that we can avoid code duplication

Comment (subhamkrai) [reply tests/github-action-helper.sh:191]: created helper function

File: tests/github-action-helper.sh (hunk 6 above)
Comment (subhamkrai) [reply tests/github-action-helper.sh]: since we created helper method and `124` is used once so I skipped creating const

File: tests/github-action-helper.sh (hunk 5 above)
Comment (subhamkrai) [reply tests/github-action-helper.sh:88]: it was missing earlier also, I added it now. Thanks

**Review by Madhu-1 — APPROVED** [review]
LGTM

**Review by sp98 — APPROVED** [review]
